Emby.Server.Implementations/Data/BaseSqliteRepository.cs

```diff
@@ -1,183 +1,76 @@
-
-        protected virtual bool EnableSingleConnection => true;
 
-        protected ManagedConnection CreateConnection(bool isReadOnly = false)
+        protected ManagedConnection GetConnection(bool _ = false)
         {
-            if (_connection != null)
+            WriteLock.Wait();
+            if (WriteConnection != null)
             {
-                return _connection;
+                return new ManagedConnection(WriteConnection, WriteLock);
             }
 
-            lock (WriteLock)
-            {
-                if (!_versionLogged)
-                {
-                    _versionLogged = true;
-                    Logger.LogInformation("Sqlite version: " + SQLite3.Version);
-                    Logger.LogInformation("Sqlite compiler options: " + string.Join(",", SQLite3.CompilerOptions.ToArray()));
-                }
-
-                ConnectionFlags connectionFlags;
-
-                if (isReadOnly)
-                {
-                    //Logger.LogInformation("Opening read connection");
-                    //connectionFlags = ConnectionFlags.ReadOnly;
-                    connectionFlags = ConnectionFlags.Create;
-                    connectionFlags |= ConnectionFlags.ReadWrite;
-                }
-                else
-                {
-                    //Logger.LogInformation("Opening write connection");
-                    connectionFlags = ConnectionFlags.Create;
-                    connectionFlags |= ConnectionFlags.ReadWrite;
-                }
-
-                if (EnableSingleConnection)
-                {
-                    connectionFlags |= ConnectionFlags.PrivateCache;
-                }
-                else
-                {
-                    connectionFlags |= ConnectionFlags.SharedCached;
-                }
-
-                connectionFlags |= ConnectionFlags.NoMutex;
+            WriteConnection = SQLite3.Open(
+                DbFilePath,
+                DefaultConnectionFlags | ConnectionFlags.Create | ConnectionFlags.ReadWrite,
+                null);
 
-                var db = SQLite3.Open(DbFilePath, connectionFlags, null);
 
-                try
-                {
-                    if (string.IsNullOrWhiteSpace(_defaultWal))
-                    {
-                        _defaultWal = db.Query("PRAGMA journal_mode").SelectScalarString().First();
-
-                        Logger.LogInformation("Default journal_mode for {0} is {1}", DbFilePath, _defaultWal);
-                    }
-
-                    var queries = new List<string>
-                    {
-                        //"PRAGMA cache size=-10000"
-                        //"PRAGMA read_uncommitted = true",
-                        "PRAGMA synchronous=Normal"
-                    };
-
-                    if (CacheSize.HasValue)
-                    {
-                        queries.Add("PRAGMA cache_size=" + CacheSize.Value.ToString(CultureInfo.InvariantCulture));
```

Comment:
why is this pragma gone?

Reply:
It was always null, it never got executed.
`protected virtual int? CacheSize => null;`

Reply:
What about the override in SqliteItemRepo:
protected override int? CacheSize => 20000;

Reply:
I'll add the logic back for the possibility to easily change this, but I thing we should do some testing to see if it's really benefiting anything. The same goes for `PRAGMA synchronous`